### DISCUSSION TITLE
Load coverage into VSCode

### DISCUSSION THREAD

**Discussion by linuxerwang** [thought-machine/please#3329] [Q&A] [answered]
Hi,

After running "plz cover", what's the correct way to let VSCode show covered/uncovered lines?

Thanks.

**Comment by chrisnovakovic** [comment, marked as answer]
Each invocation of `plz cover` writes coverage results to `plz-out/log/coverage.json` and `plz-out/log/coverage.xml` (these paths can be changed with the `--coverage_results_file` and `--coverage_xml_report` options to `plz cover`). The XML file written by Please doesn't formally conform to a DTD defining a known coverage format, but it ought to be compatible with [Cobertura's](https://github.com/cobertura/cobertura) XML format. I don't use VSCode, but from there it should be a case of using a plugin like [coverage-gutters](https://github.com/ryanluker/vscode-coverage-gutters), which apparently supports the parsing of Cobertura XML coverage files (although I make no guarantees about the suitability of that plugin).
